qa/tasks/cephfs/test_admin.py

```diff
@@ -970,6 +974,126 @@ def test_single_path_rootsquash(self):
             cmdargs = ['echo', 'some random data', Raw('|'), 'sudo', 'tee', filepath]
             self.mount_a.negtestcmd(args=cmdargs, retval=1, errmsg='permission denied')
 
+    def test_multifs_rootsquash_nofeature(self):
+        """
+        That having root_squash on one fs doesn't prevent access to others.
+        """
+
+        if not isinstance(self.mount_a, FuseMount):
+            self.skipTest("only FUSE client has CEPHFS_FEATURE_MDS_AUTH_CAPS "
+                          "needed to enforce root_squash MDS caps")
+
+        self.fs1 = self.fs
+        self.fs2 = self.mds_cluster.newfs('testcephfs2')
+
+        self.mount_a.umount_wait()
+
+        # Authorize client to fs1
+        FS_AUTH_CAPS = (('/', 'rw'),)
+        self.fs1.authorize(self.client_id, FS_AUTH_CAPS)
```

Comment:
This test failed during QA run - https://pulpito.ceph.com/xiubli-2024-07-30_09:14:39-fs-wip-xiubli-testing-20240730.051531-quincy-distro-default-smithi/7825965


```
2024-07-30T22:32:27.241 INFO:tasks.cephfs_test_runner:That having root_squash on one fs doesn't prevent access to others. ... ERROR
2024-07-30T22:32:27.241 INFO:tasks.cephfs_test_runner:
2024-07-30T22:32:27.241 INFO:tasks.cephfs_test_runner:======================================================================
2024-07-30T22:32:27.242 INFO:tasks.cephfs_test_runner:ERROR: test_multifs_rootsquash_nofeature (tasks.cephfs.test_admin.TestFsAuthorize)
2024-07-30T22:32:27.242 INFO:tasks.cephfs_test_runner:That having root_squash on one fs doesn't prevent access to others.
2024-07-30T22:32:27.242 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
2024-07-30T22:32:27.242 INFO:tasks.cephfs_test_runner:Traceback (most recent call last):
2024-07-30T22:32:27.242 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/github.com_ceph_ceph-c_449914a00d93ad72dc42c835b8e3c64d07e8025a/qa/tasks/cephfs/test_admin.py", line 993, in test_multifs_rootsquash_nofeature
2024-07-30T22:32:27.242 INFO:tasks.cephfs_test_runner:    self.fs1.authorize(self.client_id, FS_AUTH_CAPS)
2024-07-30T22:32:27.242 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/github.com_ceph_ceph-c_449914a00d93ad72dc42c835b8e3c64d07e8025a/qa/tasks/cephfs/filesystem.py", line 1612, in authorize
2024-07-30T22:32:27.242 INFO:tasks.cephfs_test_runner:    return self.get_ceph_cmd_stdout('fs', 'authorize', self.name,
2024-07-30T22:32:27.242 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/github.com_ceph_ceph-c_449914a00d93ad72dc42c835b8e3c64d07e8025a/qa/tasks/ceph_test_case.py", line 55, in get_ceph_cmd_stdout
2024-07-30T22:32:27.242 INFO:tasks.cephfs_test_runner:    return self.run_ceph_cmd(**kwargs).stdout.getvalue()
2024-07-30T22:32:27.242 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/github.com_ceph_ceph-c_449914a00d93ad72dc42c835b8e3c64d07e8025a/qa/tasks/ceph_test_case.py", line 30, in run_ceph_cmd
2024-07-30T22:32:27.242 INFO:tasks.cephfs_test_runner:    return self.mon_manager.run_cluster_cmd(**kwargs)
2024-07-30T22:32:27.242 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/github.com_ceph_ceph-c_449914a00d93ad72dc42c835b8e3c64d07e8025a/qa/tasks/ceph_manager.py", line 1631, in run_cluster_cmd
2024-07-30T22:32:27.242 INFO:tasks.cephfs_test_runner:    return self.controller.run(**kwargs)
2024-07-30T22:32:27.242 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_teuthology_53ce1462e129f6eb4071986336534c740fdebd31/teuthology/orchestra/remote.py", line 523, in run
2024-07-30T22:32:27.242 INFO:tasks.cephfs_test_runner:    r = self._runner(client=self.ssh, name=self.shortname, **kwargs)
2024-07-30T22:32:27.243 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_teuthology_53ce1462e129f6eb4071986336534c740fdebd31/teuthology/orchestra/run.py", line 447, in run
2024-07-30T22:32:27.243 INFO:tasks.cephfs_test_runner:    r = RemoteProcess(client, args, check_status=check_status, hostname=name,
2024-07-30T22:32:27.243 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_teuthology_53ce1462e129f6eb4071986336534c740fdebd31/teuthology/orchestra/run.py", line 65, in __init__
2024-07-30T22:32:27.243 INFO:tasks.cephfs_test_runner:    self.command = quote(args)
2024-07-30T22:32:27.243 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_teuthology_53ce1462e129f6eb4071986336534c740fdebd31/teuthology/orchestra/run.py", line 257, in quote
2024-07-30T22:32:27.243 INFO:tasks.cephfs_test_runner:    return ' '.join(_quote(args))
2024-07-30T22:32:27.243 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_teuthology_53ce1462e129f6eb4071986336534c740fdebd31/teuthology/orchestra/run.py", line 255, in _quote
2024-07-30T22:32:27.243 INFO:tasks.cephfs_test_runner:    yield pipes.quote(a)
2024-07-30T22:32:27.243 INFO:tasks.cephfs_test_runner:  File "/usr/lib/python3.10/shlex.py", line 329, in quote
2024-07-30T22:32:27.243 INFO:tasks.cephfs_test_runner:    if _find_unsafe(s) is None:
2024-07-30T22:32:27.243 INFO:tasks.cephfs_test_runner:TypeError: expected string or bytes-like object
2024-07-30T22:32:27.243 INFO:tasks.cephfs_test_runner:
2024-07-30T22:32:27.243 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
2024-07-30T22:32:27.243 INFO:tasks.cephfs_test_runner:Ran 25 tests in 1060.692s
```

Reply:
Will check it. Thanks @rishabh-d-dave